python-avd/pyavd/_eos_designs/structured_config/inband_management/__init__.py

```diff
@@ -83,158 +94,109 @@ def vrfs(self) -> None:
             return
         self.structured_config.vrfs.append_new(name=self.shared_utils.inband_mgmt_vrf)
 
-    @cached_property
-    def ip_virtual_router_mac_address(self) -> str | None:
+    @structured_config_contributor
+    def ip_virtual_router_mac_address(self) -> None:
         if not self.shared_utils.inband_management_parent_vlans:
-            return None
+            return
 
         if self.shared_utils.node_config.virtual_router_mac_address is None:
             msg = "'virtual_router_mac_address' must be set for inband management parent."
             raise AristaAvdInvalidInputsError(msg)
-        return str(self.shared_utils.node_config.virtual_router_mac_address).lower()
-
-    @cached_property
-    def router_bgp(self) -> dict | None:
-        if not self.shared_utils.inband_management_parent_vlans:
-            return None
+        self.structured_config.ip_virtual_router_mac_address = str(self.shared_utils.node_config.virtual_router_mac_address).lower()
```

Comment:
I don't think you need the str() any more. The type hint on the class should know is is a string.

Reply:
Fixed!